lib/pkg.mli

```diff
@@ -154,7 +149,7 @@ val build : f
 
 (** {1 Version} *)
 
-val extract_version : t -> (string, Sos.error) result
+val extract_version : t -> (Version.t, Sos.error) result
```

Comment:
Same as I mentioned above but this should probably return a string to avoid confusion. Using a `Version.t` is confusing as I wouldn't expect a `Version.t` to contain the leading `v` if `keep-v` is `false` for example.

Reply:
Might be worth extracting a new type while we're at it. Might be a `Changelog_version.t` or `Version.Changelog.t`. That would also prevent us from badly converting from this type to `Tag.t` or `Version.t`. What do you think?